rewrite-groovy/src/main/java/org/openrewrite/groovy/GroovyParserVisitor.java

```
@@ -403,17 +434,27 @@ public void visitBlockStatement(BlockStatement statement) {
 
         @Override
         public void visitField(FieldNode field) {
-            if ((field.getModifiers() & Opcodes.ACC_ENUM) != 0) {
-                visitEnumField(field);
-            } else {
-                visitVariableField(field);
+            if (field.isEnum()) {
+                // Enum constants are handled separately in visitClassBlock, thus should be skipped here
+                return;
             }
+            visitVariableField(field);
         }
 
-        private void visitEnumField(@SuppressWarnings("unused") FieldNode fieldNode) {
-            // Requires refactoring visitClass to use a similar pattern as Java11ParserVisitor.
-            // Currently, each field is visited one at a time, so we cannot construct the EnumValueSet.
-            throw new UnsupportedOperationException("enum fields are not implemented.");
+        private J.EnumValue visitEnumField(FieldNode field) {
+            Space prefix = whitespace();
+
+            List<J.Annotation> annotations = visitAndGetAnnotations(field, this);
+
+            Space namePrefix = whitespace();
+            String enumName = field.getName();
+            skip(enumName);
+
+            J.Identifier name = new J.Identifier(randomId(), namePrefix, Markers.EMPTY, emptyList(), enumName, typeMapping.type(field.getType()), typeMapping.variableType(field));
+
+            // TODO initializer (enum constructor invocation)
```

Instead of TODO wouldn't it be better to throw `UnsupportedOperationException` when a non-empty initializer exists? Otherwise we might be masking implementation gaps.